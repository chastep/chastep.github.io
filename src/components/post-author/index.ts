export { PostAuthor } from "./post-author";
