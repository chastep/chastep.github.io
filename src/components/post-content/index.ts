export { PostContent } from "./post-content";
