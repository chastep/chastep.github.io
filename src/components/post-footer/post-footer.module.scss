@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.post-footer {
  .date {
    font-style: italic;
  }
}
