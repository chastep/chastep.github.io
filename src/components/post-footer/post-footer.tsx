import React, { type FC } from "react";

import * as styles from "./post-footer.module.scss";

interface PostFooterProps {
  date: string;
}

const PostFooter: FC<PostFooterProps> = ({ date }) => (
  <div className={styles.postFooter}>
    <p className={styles.date}>
      Published{" "}
      {new Date(date).toLocaleDateString("en-US", {
        year: "numeric",
        month: "short",
        day: "numeric",
      })}
    </p>
  </div>
);

export { PostFooter };
