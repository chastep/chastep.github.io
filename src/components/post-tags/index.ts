export { PostTags } from "./post-tags";