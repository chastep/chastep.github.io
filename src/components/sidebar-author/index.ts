export { SidebarAuthor } from "./sidebar-author";