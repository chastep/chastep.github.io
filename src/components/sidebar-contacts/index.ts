export { SidebarContacts } from "./sidebar-contacts";
