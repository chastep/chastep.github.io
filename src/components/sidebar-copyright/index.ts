export { SidebarCopyright } from "./sidebar-copyright";
