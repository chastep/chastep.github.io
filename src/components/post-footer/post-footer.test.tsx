import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { PostFooter } from "@/components/post-footer";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("PostFooter", () => {
  test("renders correctly", () => {
    const props = { date: mocks.markdownRemark.frontmatter.date };
    const tree = createSnapshotsRenderer(<PostFooter {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
