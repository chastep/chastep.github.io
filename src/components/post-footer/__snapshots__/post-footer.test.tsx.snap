// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFooter renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "Published",
        " ",
        "Sep 1, 2016",
      ],
      "props": {
        "className": undefined,
      },
      "type": "p",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
